lib/quicly.c

```diff
@@ -3345,6 +3494,53 @@ static int send_connection_close(quicly_conn_t *conn, quicly_send_context_t *s)
     return 0;
 }
 
+static int send_new_connection_id(quicly_conn_t *conn, quicly_send_context_t *s, struct st_quicly_pending_new_cid_t *new_cid)
+{
+    int ret;
+    quicly_cid_plaintext_t plain_cid = conn->super.master_id;
+    quicly_cid_t cid;
+    uint8_t stateless_reset_token[QUICLY_STATELESS_RESET_TOKEN_LEN];
+    quicly_sent_t *sent;
+    quicly_cid_encryptor_t *cid_encryptor = conn->super.ctx->cid_encryptor;
+    uint64_t retire_prior_to = 0; /* TODO */
+
+    plain_cid.path_id = new_cid->path_id;
+
+    assert(cid_encryptor != NULL);
+    cid_encryptor->encrypt_cid(cid_encryptor, &cid, stateless_reset_token, &plain_cid);
```

Comment:
You're totally right. I somehow (incorrectly) assumed that `encrypt_cid` was purely functional. Thank you for catching it.

Reply:
Done.

Reply:
Coming back on this topic. Given `cid_encryptor` may change over time, I wonder if we could have a scenario like this:
1. server issues multiple CIDs upon connection establishment
2. cid_encryptor changes in the server process
3. client switches to one of the CIDs issued in step 1
4. server tries to decrypt the CID but since cid_encryptor is different from what it had when the CID was issued (step 1), it cannot correctly obtain `cid_plaintext_t`, and things like `quicly_is_destination` don't work as expected.

Is this something that could actually happen, and if so, is it something to worry about? Maybe the probability of hitting such a case is low enough?